test-data/unit/check-isinstance.test

```diff
@@ -2573,3 +2573,27 @@ if issubclass(x, B):
 else:
     reveal_type(x)        # N: Revealed type is "Type[__main__.A]"
 [builtins fixtures/isinstance.pyi]
+
+[case testTypeEqualsCheck]
+from typing import Any
+
+y: Any
+if type(y) == int:
+    reveal_type(y) # N: Revealed type is "builtins.int"
+
+
+[case testMultipleTypeEqualsCheck]
+from typing import Any
+
+x: Any
+y: Any
+if type(x) == type(y) == int:
+    reveal_type(y) # N: Revealed type is "builtins.int"
+    reveal_type(x) # N: Revealed type is "builtins.int"
+
+[case testTypeEqualsCheckUsingIs]
+from typing import Any
+
+y: Any
+if type(y) is int:
+    reveal_type(y) # N: Revealed type is "builtins.int"
```

Comment:
Add also test case where you narrow from `Union[int, str]` to `int`, and there is an else block. For example, something like this:

```py
def f(x: Union[int, str]) -> None:
    if type(x) is int:
        reveal_type(x)  # int
    else:
        reveal_type(x)  # Union[int, str]
```

The else block should use the original type, since instances of `int` subclasses (e.g. `bool`) will be handled in the else block.

Reply:
Ok, I have added a test case for this and modified the type checker to handle this. Could you take a look at my changes when you get the chance?